src/Api/AuthAction.php

```diff
@@ -19,27 +19,49 @@
 
 class AuthAction implements Action
 {
+    /**
+     * @var SettingsRepository
+     */
     protected $settings;
 
+    /**
+     * @var Pusher
+     */
+    protected $pusher;
+
     public function __construct(SettingsRepository $settings)
     {
         $this->settings = $settings;
+
+        $this->pusher = new Pusher(
+            $this->settings->get('pusher.app_key'),
+            $this->settings->get('pusher.app_secret'),
+            $this->settings->get('pusher.app_id')
+        );
```

Comment:
Hmm, this shouldn't have been done here. The action should not need to know how to instantiate the pusher connection.

I realize you just copied this, but can we move this to the service provider?